data/binding/gen.go

```diff
@@ -130,25 +132,26 @@ func {{ .Name }}ToStringWithFormat(v {{ .Name }}, format string) String {
 	return str
 }
 
-func (s *stringFrom{{ .Name }}) Get() string {
-	val := s.from.Get()
+func (s *stringFrom{{ .Name }}) Get() (string, error) {
+	val, err := s.from.Get()
 
-	return fmt.Sprintf(s.format, val)
+	return fmt.Sprintf(s.format, val), err
 }
 
-func (s *stringFrom{{ .Name }}) Set(str string) {
+func (s *stringFrom{{ .Name }}) Set(str string) error {
 	var val {{ .Type }}
 	n, err := fmt.Sscanf(str, s.format, &val)
 	if err != nil || n != 1 {
-		fyne.LogError("{{ .Type }} parse error", err)
-		return
+		return err
 	}
-	if val == s.from.Get() {
-		return
+	old, err := s.from.Get()
+	if val == old {
+		return err
```

Comment:
This is combining two cases, and it would be better to be more explicit;
```go
old, err := s.from.Get()
if err != nil {
    return err
}
if val == old {
    return nil
}
```